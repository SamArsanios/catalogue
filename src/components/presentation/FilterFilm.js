import React from 'react';
import PropTypes from 'prop-types';

const FilterFilm = (props) => {
  const { handleFilterChange } = props;
  const genres = ['All', 'movie', 'series'];

  return (
    <div>
      <select
        onChange={handleFilterChange}
        className="form-select my-5"
        aria-label="Default select example"
      >
        {
          genres.map((genre) => (
            <option
              key={genre}
              value={genre}
            >
              {genre}
            </option>
          ))
        }
      </select>
    </div>
  );
};

FilterFilm.propTypes = {
  handleFilterChange: PropTypes.func.isRequired,
};

export default FilterFilm;
