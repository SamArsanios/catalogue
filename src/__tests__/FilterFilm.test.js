import { BrowserRouter } from 'react-router-dom';
import { create } from 'react-test-renderer';
import FilterFilm from '../components/presentation/FilterFilm';

test('renders correctly', () => {
  const Placeholder = () => (
    <BrowserRouter>
      <FilterFilm />
    </BrowserRouter>
  );
  const tree = create(<Placeholder />).toJSON();
  expect(tree).toMatchSnapshot();
});
