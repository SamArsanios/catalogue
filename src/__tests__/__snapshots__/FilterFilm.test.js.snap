// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <select
    aria-label="Default select example"
    className="form-select my-5"
    onChange={[Function]}
  >
    <option
      value="All"
    >
      All
    </option>
    <option
      value="movie"
    >
      movie
    </option>
    <option
      value="series"
    >
      series
    </option>
  </select>
</div>
`;
